platform/ecs/instance/client.go

```diff
@@ -68,6 +69,11 @@ func (c *client) newRequest(endpoint string) (*http.Request, error) {
 }
 
 func readBody(resp *http.Response) ([]byte, error) {
-	defer resp.Body.Close()
+	if resp.Body != nil {
+		defer resp.Body.Close()
+	}
+	if resp.StatusCode != http.StatusOK {
+		return []byte{}, fmt.Errorf("incorrect status code %d", resp.StatusCode)
```

Comment:
Why returns `[]byte{}` instead of `nil`?

Reply:
I was mistaken about it.